Fix column key sort with col block starting in col1 bug

sasumner opened the pull request:
Fixes #8716

Note that when a column block begins in (user) column 1, `_fromColumn` is zero; in the original code this would cause `isSortingSpecificColumns()` to return `false`.

donho commented:
@sasumner 
Isn't `return !( _fromColumn == 0 && _toColumn == 0);` more explicit?
Just want to have your thought.

sasumner commented:
@donho

>Isn't return !( _fromColumn == 0 && _toColumn == 0); more explicit?

I thought about doing it that way.
It seems more complicated but it probably reads better.

I think this would also work:
`return _toColumn != 0;`

Let me know if I should change it to your version; it's fine.

sasumner commented:
Implemented the `return _toColumn !=0;`.

Also, moved an unrelated comment slightly to have it make more sense.
Should have been moved with #8674 changes but wasn't.
I hope it is OK to move that comment here.

donho commented:
@sasumner 
I'm not a psychopath against to all "unrelated" modification. I'm OK with all modification in the right direction, if it doesn't prevent me from code review. :)
